ignite/engine/engine.py

```diff
@@ -706,23 +708,23 @@ def switch_batch(engine):
         return self._internal_run()
 
     @staticmethod
-    def _init_timers(state: State):
+    def _init_timers(state: State) -> None:
         state.times[Events.EPOCH_COMPLETED.name] = 0.0
         state.times[Events.COMPLETED.name] = 0.0
 
-    def _get_data_length(self, data):
-        data_length = None
+    def _get_data_length(self, data: Optional[Iterable] = None) -> Optional[int]:
```

Comment:
I'm not sure about `Optional[Iterable]` here, I think `data` can not be None here.

Reply:
Yeah, you are right. I retraced the usage of `_setup_engine()` and some hops back, there is an assignment of `dataloader` with an not `None` value. As it seems, it is too hard for `mypy` to figure that out. Then I will do a `None` check and `raise` an error.